utils/walk.js

```diff
@@ -0,0 +1,45 @@
+const bcd = require('./bcd');
+const { isBrowser, descendantKeys, joinPath } = require('./walkingUtils');
+const query = require('./query');
+
+function* lowLevelWalk(data = bcd, path, depth = Infinity) {
```

Comment:
Why two walking generators? `walk()` wraps `lowLevelWalk()` but only yields `__compat: { … }` objects. `lowLevelWalk()` is the basis for other sorts of interesting walks, such as walking non-`__compat` namespaces or walking browsers, which have not been implemented yet.